src/uct/api/uct.h

```diff
@@ -1309,21 +1309,27 @@ typedef struct uct_md_mem_attr {
 
     /**
      * The type of memory. E.g. CPU/GPU memory or some other valid type
+     * If the md does not support sys_dev query, then UCS_MEMORY_TYPE_UNKNOWN
+     * is returned.
      */
     ucs_memory_type_t mem_type;
 
     /**
      * Index of the system device on which the buffer resides. eg: NUMA/GPU
+     * If the md not support sys_dev query, then UCS_SYS_DEVICE_ID_UNKNOWN
+     * is returned.
      */
     ucs_sys_device_t  sys_dev;
 
     /**
      * Base address of the allocation to which the provided buffer belongs to.
+     * If the md not support base address query, then NULL is returned.
```

Comment:
should return same address as passed

Reply:
Concerning the above case and this https://github.com/openucx/ucx/pull/6178/#discussion_r564801229. What about the following case?
```
a = malloc(4096); // or cudaMalloc for that matter
uct_md_mem_query(..., address=(a + 32), length=32, ...)
```
It doesn't seem right that mem_attr.base_address return `a + 32` and mem_attr.alloc_length return `32`

Reply:
@yosefe how should we proceed here?

Reply:
why doesn't it make sense? it will simplify the code which tries to "expand" the memory registration

Reply:
If registration is the only user of uct_md_mem_query with BASE_ADDR, ALLOC_LENGTH attributes requested, then the suggestion to return the pointer and length as is is not erroneous. But if there are other users, then isn't it misleading? Even for registration, ideally we want to return `a` as based address and `4096` (or greater in case the actual allocation results in a larger allocation than user requested) otherwise we're not expanding when there's an option to do so.

Reply:
i think the notion of "base address" is not well defined enough to require that base address/length are page multiples.
it should be defined more loosely, as a "continuous memory region which contains address/length and has same memory properties". IMO, anything more strict than that will be hard to define in a transport-independent way

Reply:
That makes sense. I'll make changes as per your suggestion.

Reply:
thanks, let's also document this in the API.

Reply:
Addressed this.